worker/uniter/resolver.go

```diff
@@ -304,6 +305,13 @@ func (s *uniterResolver) nextOp(
 ) (operation.Operation, error) {
 	switch remoteState.Life {
 	case life.Alive:
+		if remoteState.Shutdown {
+			if localState.Started && !localState.Stopped {
+				return opFactory.NewRunHook(hook.Info{Kind: hooks.Stop})
+			} else if !localState.Started || localState.Stopped {
+				return nil, worker.ErrTerminateAgent
+			}
```

Comment:
Just checking - this covers the install case properly? eg we've run the storage attached hooks and install but not started yet